lumicks/pylake/piezo_tracking/piezo_tracking.py

```diff
@@ -0,0 +1,166 @@
+        coeffs = np.polyfit(self.position, self.distance, degree)
+        self._model = np.poly1d(coeffs)
+
+    def __call__(self, trap_position):
+        return Slice(
+            trap_position._src._with_data(self._model(trap_position.data)),
+            labels={"title": "Piezo distance", "y": "Distance [um]"},
+        )
+
+    def valid_range(self):
+        return (np.min(self.position), np.max(self.position))
+
+    def __str__(self):
+        powers = np.flip(np.arange(self._model.order + 1))
+        return "".join(
+            f"{' + ' if coeff > 0 else ' - '}"
+            f"{abs(coeff):.4f}"
+            f"{'' if power == 0 else ' x' if power == 1 else f' x^{power}'}"
+            for power, coeff in zip(powers, self._model.coeffs)
+        ).strip()
+
+    def __repr__(self):
+        return f"DistanceCalibration({str(self)})"
+
+    def plot(self):
+        """Plot the calibration fit"""
+        plt.scatter(self.position, self.distance, s=2, label="data")
+        plt.plot(self.position, self._model(self.position), "k", label=f"${str(self)}$")
+        plt.xlabel("Mirror position")
+        plt.ylabel("Camera Distance [um]")
+        plt.tight_layout()
+        plt.legend()
+
+    def plot_residual(self):
+        """Plot the residual of the calibration fit"""
+        plt.scatter(self.position, self._model(self.position) - self.distance, s=2)
+        plt.ylabel("Residual [um]")
+        plt.xlabel("Mirror position")
+        plt.tight_layout()
+        plt.legend()
+
+    @classmethod
+    def from_file(cls, calibration_file, degree=1):
+        """Use a reference measurement to calibrate trap mirror position to bead-bead distance.
+
+        Parameters
+        ----------
+        calibration_file : pylake.File
+        degree : int
+            Polynomial order.
+        """
+        return cls(calibration_file["Trap position"]["1X"], calibration_file.distance1, degree)
+
+
+class PiezoTrackingCalibration:
+    def __init__(
+        self,
+        trap_calibration,
+        baseline_force1,
+        baseline_force2,
+        signs=(1, -1),
+    ):
+        """Set up piezo tracking calibration
+
+        trap_calibration : pylake.DistanceCalibration
+            Calibration from trap position to trap to trap distance.
+        baseline_force1 : pylake.ForceBaseline
+            Baseline for force1
+        baseline_force2 : pylake.ForceBaseline
+            Baseline for force2
+        signs : tuple(float, float)
+            Sign convention for forces (e.g. (1, -1) indicates that force2 is negative).
+        """
+        if len(signs) != 2:
+            raise ValueError(
+                "Argument `signs` should be a tuple of two floats reflecting the sign for each "
+                "channel."
+            )
+        for sign in signs:
+            if abs(sign) != 1:
+                raise ValueError("Each sign should be either -1 or 1.")
+
```

Comment:
It would be good to give a warning if the user does not provide a trap_calibration and baseline_force as input.
It is eg possible to create a PiezoTrackingCalibration object using the trap position.

Reply:
Yeah, it's a fair point to check it in this case, since it wouldn't fail at the point where this object was created but later, which could indeed be confusing. I've added some type validation here.